flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartition.java

```diff
@@ -167,6 +167,13 @@ public int releaseMemory() {
 	}
 
 	@Override
+	public boolean isReleased() {
```

Comment:
I think it makes no difference to synchronize around variable access. A variable read is atomic anyways, the synchronization does not add anything there...